minio/console
# Use Console as proxy for share object logic

solves: https://github.com/minio/console/issues/3255

Share File URL now points to Console to facilitate sharing the url in case MinIO endpoint is not exposed.

Changes:
- Update how we return the share url pointing now to Object Browser API instead of MinIO endpoint
- Adds new endpoint to validate share link url (base64encoded) and download file
- unittests to validate url

How it looks:
![Screenshot 2024-04-05 at 2 49 32 PM](https://github.com/minio/console/assets/11819101/9e318bb8-98d5-4bef-b128-f9eb5fd8a56a)

When url expired:
![Screenshot 2024-04-05 at 2 50 58 PM](https://github.com/minio/console/assets/11819101/c541f170-27b7-41ce-92ea-ae0522dd7d01)

### Test steps:
With an existing bucket, folder and objects.

- Click share file
- Copy and paste the new url in the browser
- The url should contain same url as Object browser
- Object should be downloaded
- Test injecting wrong urls (base64encoded)
- Should return error or forbidden errors if url doesn't point to internal MinIO server
